test/tuple.jl

```diff
@@ -150,11 +150,11 @@ end
     @test_throws BoundsError getindex((1,2), 0)
     @test_throws BoundsError getindex((1,2), -1)
 
-    @test getindex((1,), 1.0) === 1
-    @test getindex((1,2), 2.0) === 2
-    @test_throws BoundsError getindex((), 1.0)
-    @test_throws BoundsError getindex((1,2), 0.0)
-    @test_throws BoundsError getindex((1,2), -1.0)
+    @test_deprecated getindex((1,), 1.0) === 1
+    @test_deprecated getindex((1,2), 2.0) === 2
+    @test_throws Exception getindex((), 1.0)
+    @test_throws Exception getindex((1,2), 0.0)
+    @test_throws Exception getindex((1,2), -1.0)
```

Comment:
Probably these should be moved to https://github.com/JuliaLang/julia/blob/master/test/deprecation_exec.jl (which doesn't run with `--depwarn=error`).